CWE-664/CWE-833/README.md

```diff
@@ -0,0 +1,241 @@
+CWE-833: Deadlock
+====================
+
+The number of threads that can simultaneously run within a `ThreadPoolExecutor` is limited by the `_max_workers` parameter. Submitting tasks whose execution is dependent on other tasks submitted to the same `ThreadPoolExecutor` may result in a *thread-starvation* deadlock. An example of this type of deadlock is shown in the following article [Brownlee, 2021], which describes it as "Deadlock 1: Submit and Wait for a Task Within a Task". Submitting a task will add it to an internal `ThreadPoolExecutor` queue. The task will be removed from the queue when one of the worker threads becomes available, i.e., after finishing its current task. If all workers are busy and all their current tasks are waiting for results from tasks that are waiting in the queue, the program will run indefinitely as no worker will be able to complete its task and take one of the blocking tasks from the queue.
```

Comment:
Well our original decision was to use the  Citing and referencing material - Harvard referencing quick guide - LibGuides at Dundalk Institute of Technology 
https://dkit.ie.libguides.com/harvard/citing-referencing

Reply:
Hi @gkunz, talked to the team.  We like to get this solved but not at this time as it would delay publishing the content. Are you ok if we continue as is without a solution? This will all become much easier to fix once we have everything in github.